### PR TITLE
Run validation subtests with testing.Run()

### PR DESCRIPTION
**Pull request by luksa**
NOTE for reviewers: the changes are much simpler than they look in GitHub's diff viewer. All I did was wrap code in `t.Run()`. I suggest using GoLand IDE's diff viewer (or similar).

### CODE REVIEW

**Comment by MHBauer** [comment]
looks like #2185

File: pkg/apis/servicecatalog/validation/broker_test.go [hunk 1]
```diff
@@ -519,13 +519,14 @@ func TestValidateClusterServiceBroker(t *testing.T) {
...
-		if len(errs) != 0 && tc.valid {
-			t.Errorf("%v: unexpected error: %v", tc.name, errs)
-			return
-		} else if len(errs) == 0 && !tc.valid {
-			t.Errorf("%v: unexpected success", tc.name)
-		}
+		t.Run(tc.name, func(t *testing.T) {
+			errs := ValidateClusterServiceBroker(tc.broker)
+			if len(errs) != 0 && tc.valid {
+				t.Errorf("%v: unexpected error: %v", tc.name, errs)
```
Comment (nilebox) [inline pkg/apis/servicecatalog/validation/broker_test.go]: No need to print `tc.name` anymore?

File: pkg/apis/servicecatalog/validation/broker_test.go [hunk 2]
```diff
@@ -519,13 +519,14 @@ func TestValidateClusterServiceBroker(t *testing.T) {
...
+			} else if len(errs) == 0 && !tc.valid {
+				t.Errorf("%v: unexpected success", tc.name)
```
Comment (nilebox) [inline pkg/apis/servicecatalog/validation/broker_test.go]: `tc.name` is redundant

Comment (jberkhahn) [reply pkg/apis/servicecatalog/validation/broker_test.go]: This is redundant, but I actually prefer the output this generates as it's the actual text that's control-f'able to find the actual failing test.

For reference, this is what the generated output looks like:
current PR:
```
jberkhahn@Purgatory> go test .
--- FAIL: TestValidateClusterServiceBroker (0.00s)
    --- FAIL: TestValidateClusterServiceBroker/valid_clusterservicebroker_-_catalogRequirements_with_serviceClass_and_servicePlan (0.00s)
    	broker_test.go:527: valid clusterservicebroker - catalogRequirements with serviceClass and servicePlan: unexpected success
FAIL
FAIL	github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/validation	0.097s
```
with references to tc.name removed from the Errorf calls:
```
jberkhahn@Purgatory> go test .
--- FAIL: TestValidateClusterServiceBroker (0.00s)
    --- FAIL: TestValidateClusterServiceBroker/valid_clusterservicebroker_-_catalogRequirements_with_serviceClass_and_servicePlan (0.00s)
    	broker_test.go:527: unexpected success
FAIL
FAIL	github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/validation	0.123s

```

Comment (nilebox) [reply pkg/apis/servicecatalog/validation/broker_test.go]: @jberkhahn I still prefer the output with `tc.name` removed, it's easy to see the test case just one line above.

File: pkg/apis/servicecatalog/validation/broker_test.go (hunk 1 above)
Comment (luksa) [reply pkg/apis/servicecatalog/validation/broker_test.go]: Oops, yeah, I didn't notice that. I've now removed it everywhere.

**Pull request merged**